dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsWriter.java

```diff
@@ -0,0 +1,135 @@
+            saxParser = factory.newSAXParser();
+        } catch (ParserConfigurationException e) {
+            throw new RuntimeException(e);
+        } catch (SAXException e) {
+            throw new RuntimeException(e);
+        }
+        TestSuiteXmlParser xmlParser = new TestSuiteXmlParser();
+
+        try (FileWriter output = new FileWriter(outputFile)) {
+            Path junitPath = Paths.get("results", "junit");
+            File junitDirectory = junitPath.toFile();
+            for (final File xmlFile : junitDirectory.listFiles()) {
+                if (xmlFile.isDirectory()) {
+                    continue; //this shouldn't happen.
+                }
+                if (xmlFile.length() == 0) {
+                    continue; //this probably will happen. We create an empty file then populate it after we're expecting this method to run.
+                }
+                saxParser.parse(xmlFile, xmlParser);
+            }
+
+            String adocContent = "";
+            String MPversion = "";
+
+            String MPSpecLower = (specName.toLowerCase()).replace(" ", "-");
+            String rcVersion = specParts[2];
+            String[] documentStart = { ":page-layout: certification \n= TCK Results\n\n",
+                                       "As required by the https://www.eclipse.org/legal/tck.php[Eclipse Foundation Technology Compatibility Kit License], following is a summary of the TCK results for releases of ",
+                                       type //MicroProfile or Jakarta EE
+            };
+            for (String part : documentStart) {
+                adocContent += part;
+            }
+            if (type.equals("MicroProfile")) {
+                String[] documentMain = { " ", specName, " ", specVersion,
+                                          ".\n\n== Open Liberty ", openLibertyVersion, " - MicroProfile ", specName, " ", specVersion,
+                                          " Certification Summary (Java ", javaMajorVersion, ")\n\n",
+                                          "* Product Name, Version and download URL (if applicable):\n",
+                                          "\nhttps://repo1.maven.org/maven2/io/openliberty/openliberty-runtime/",
+                                          openLibertyVersion, "/openliberty-runtime-", openLibertyVersion, ".zip[Open Liberty ", openLibertyVersion, "]\n",
+                                          "* Specification Name, Version and download URL:\n\n",
+                                          "link:https://download.eclipse.org/microprofile/microprofile-", MPSpecLower, "-", specVersion, rcVersion,
+                                          "/microprofile-", MPSpecLower, "-spec-", specVersion, rcVersion, ".html[MicroProfile ", specName, " ", specVersion,
+                                          rcVersion, "]\n\n* Public URL of TCK Results Summary:\n\n", "link:", filename, "[TCK results summary]\n\n",
```

Comment:
In our existing files, the link here ends with `.html`, whereas with this change, `filename` will end with `.adoc`.

I'm unsure whether this difference will be handled when the file is rendered, but we need to check and change it if necessary.

Reply:
Possibly we should use `xref` rather than `link`: https://docs.asciidoctor.org/asciidoc/latest/macros/inter-document-xref/